andela/travel_tool_front
# #162474402 Display proper 404 message when content does not exist is returned

#### What does this PR do?
- Renders 404 error messages fro pages that return content does not exist
- Fixes potential bugs as a result of rendering no content when you hit the verifications route via URL link.

#### Description of Task to be completed?
- Renders 404 error messages fro pages that return content does not exist

#### How should this be manually tested?
- Setting up the backend
     - Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
     - Enter into the project directory - `cd travel_tool_back`
     - checkout the branch git checkout `develop`
     - Install dependencies using `yarn install`
     - Rollback and run database migrations with `yarn db:rollmigrate`
     - start the backend server with `yarn run start:dev`


- Setting up the frontend
   - Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
   - Enter into the project directory - `cd travel_tool_front`
   - Checkout to the branch `ft-no-content-components-162474402` then run the app `yarn start`
   - Login to the application
   - As you change your User roll, navigate to any of these routes:
       - `/requests/fake-Id`
       - `/requests/my-approvals/fake-Id`
       - `/requests/my-verifications/fake-Id`
       - `/residence/manage/fake-Id`
       - `/settings/roles/fake-Id`
   - It is expected that the page not found component will be rendered and the user will naviagate back to the previous page if they use the link provided.

#### Any background context you want to provide?
- By default, if the Id of the travel request exists in the URL, set the state but do not open the Modal until the requests/ approvals/ verifications have been fetched. When the data has been fetched and the Id exists within the results as in `componentDidUpdate` then open the Modal.

#### What are the relevant pivotal tracker stories?
[162474402](https://www.pivotaltracker.com/story/show/162474402)

#### Screenshots (if appropriate)
![ezgif com-optimize 1](https://user-images.githubusercontent.com/21138053/49836011-2e718e00-fdb2-11e8-806a-6e1eafa9b2f4.gif)
